### PR TITLE
[top_englishbreakfast] Remove self-vendoring dependency

### PR DESCRIPTION
**Pull request by mcy**
Fixes https://github.com/lowRISC/opentitan/issues/8007. The current fix is... a different hack, admittedly, but one that is much less fragile and suggests a way to move forward with a custom Bazel platform (in `opentitan.bzl`). In the sequel, I'll figure out how we want to spell this in Bazel, probably via a `-c` flag and some `select()`s.

### CODE REVIEW

**Review by cfrantz — APPROVED** [review]

**Review by timothytrippel — APPROVED** [review]
This is FANTASTIC! Been waiting for this day hehe.

File: meson.build [hunk 1]
```diff
@@ -19,6 +19,9 @@ project(
   ],
 )
 
+# NOTE: There are tools in hw/top_englishbreakfast that mess with this line.
```
Comment (vogelpi) [inline meson.build:22]: Would you mind to change the language here a bit? It's perfectly sufficient to say that these tools "need" this line.

File: hw/top_englishbreakfast/util/prepare_sw.py [hunk 2]
```diff
@@ -124,21 +124,6 @@ def main():
         with open(path_out + file_name_new, "w") as file_out:
             file_out.write(text)
 
-    # 4.
```
Comment (vogelpi) [inline hw/top_englishbreakfast/util/prepare_sw.py:127]: Could you please also remove Lines 80 and 81 of this file? These comments outline why this block No. 4 here is needed and what it does.

**Review by vogelpi — APPROVED** [review]
Thanks a lot @mcy for doing this change! I think the macro is a much better way to handle these differences between the top levels.

Ironically, when I started working on the English Breakfast toplevel there was heavy opposition against using macro anywhere in the code base. As I was more or less on my own working on this, I ended up adding the hacky patch file. Thankfully, this is now over :-)

**Pull request merged**